libs/dal/src/repositories/notification-template/notification-template.repository.ts

```diff
@@ -46,22 +46,56 @@ export class NotificationTemplateRepository extends BaseRepository<
     const requestQuery: NotificationTemplateQuery = {
       _id: id,
       isBlueprint: true,
-      _organizationId: NotificationTemplateRepository.getBlueprintOrganizationId() as string,
+      _environmentId: NotificationTemplateRepository.getBlueprintEnvironmentId() as string,
     };
 
     const item = await this.MongooseModel.findOne(requestQuery).populate('steps.template');
 
     return this.mapEntity(item);
   }
 
+  async findAllGroupedByCategory(): Promise<{ name: string; blueprints: NotificationTemplateEntity[] }[]> {
+    if (!NotificationTemplateRepository.getBlueprintEnvironmentId()) {
```

Comment:
```suggestion
   const blueprintId = NotificationTemplateRepository.getBlueprintEnvironmentId();
    if (!blueprintId) {
```
We need the `blueprintId` further in this method so we can reuse the value.